### ISSUE TITLE
Force bad scrollbar designs to apply accessibility patches

### ISSUE THREAD

**Issue by summercms** [openui/open-ui#404]
### Background

![1_kh08HW91JyQ2FMUyHGRA9w](https://user-images.githubusercontent.com/57409060/134156771-d29e9122-43ed-4f10-9368-0ffcdb226371.png)

Website developers/designers can now customize the scrollbar in the browser, using this spec: https://drafts.csswg.org/css-scrollbars-1/

Browser support is now over 97% https://caniuse.com/?search=scrollbar

### Issue

I have come across several websites that have customized the browser scrollbar. But their websites have really bad **Scroll Jank** issues!

> "Jank" is the term used to describe the visual hiccup that occurs when the system is not able to build and provide frames in time for them to be drawn to the screen at the requested cadence (60hz, or higher). Jank is most apparent when scrolling, when what should be a smoothly animated flow has hiccups, where the movement pauses along the way for one or more frames as the app takes longer to render content than the duration of a frame on the system.

Because of the JANK issues; when I try to use the `mouse wheel` to scroll down the webpage. The `jank` is preventing the page from scrolling down. So I try to grab the scrollbar with the `mouse pointer` and move the scrollbar down.

However, I have seen quite a few websites with really thin scrollbars that I end up spending several minutes trying to grab a scrollbar that is say 1-3px wide for example.

![image](https://user-images.githubusercontent.com/57409060/134551853-cfdbb142-a194-490c-bece-f3c4a7ee0837.png)

### Solution

The solution would be for browser vendors to transform the scrollbar width to the normal width (or a width of a minimum set standard) when a user hovers the mouse pointer over the scrollbar. This way people with accessibility issues (such as myself - who has Osteoarthritis in my hands) can easily grab customized scrollbars that are so thin it would be very tricky to click on them with the mouse pointer needing to grad and scroll the browser scrollbar on a computer!

Example:

![scrollbar](https://user-images.githubusercontent.com/57409060/134160275-f9b5e649-c150-4ff2-9f3f-d0f901481e68.png)

Above when a user moves the mouse pointer over the customized scrollbar the width is automatically increased to make it easy for users to click and grab the scrollbar (when the scrollbar has been styled too thin on a set predefined width).

### Open-ui

Can create some customized browser scrollbar examples that have accessibility patches built in, that developers can use. Also create a spec that incorporates customize scrollbar css and accessibility standards.

Extends this issue: https://github.com/openui/open-ui/issues/8

### Browsers

I have noticed browsers like `Chrome` and `Edge` have started adding some proposals from this repo into their browser designs and I hope Browsers can add this enhancement to patch poorly designed scrollbars that have accessibility issues!

**Comment by gregwhitworth** [comment]
I really like that you brought this here. @tantek since you drove this in CSS can you please weigh in if you'd like to have it here or move this issue over to the CSSWG. I'll add this to the agenda of the telecon solely to discuss and get agreement on where to investigate this. The ONLY reason I am mixed here is due to so many widgets having scrollers that we need to incorporate in some way.

**Comment by tantek** [comment]
The [https://drafts.csswg.org/css-scrollbars-1/#scrollbar-width](https://drafts.csswg.org/css-scrollbars-1/#scrollbar-width) definition for the 'thin' value already explicitly asks for the requested behavior: "remain wide enough to be usable" and cites WCAG accordingly.

If folks have suggestions for better / stronger language or citations that we could use, please feel free to file an issue on the Scrollbars Styling draft at [https://github.com/w3c/csswg-drafts/labels/css-scrollbars-1](https://github.com/w3c/csswg-drafts/labels/css-scrollbars-1) with suggestions!

Thanks,

Tantek (Scrollbars Styling co-editor)

(Originally published at: https://tantek.com/2021/266/t1)

**Comment by gregwhitworth** [comment]
@summercms can you provide the site that you were testing this on as I'd like to look at it and which browser this is in relation to.

I'll file a CSSWG issue to change the terminology to "MUST" rather than what is currently authored as the browser should be able to handle it.

**Comment by summercms** [comment]
@gregwhitworth 

> can you provide the site that you were testing this on as I'd like to look at it and which browser this is in relation to.

I seem to have lost the website in my long history list (it was a major news website). But everyday I find more websites with this issue and today I have come across another website example.

Link: https://www.tradingview.com/brokers/

Click the search box and you will see the following:

![image](https://user-images.githubusercontent.com/57409060/134674940-82880b6f-361d-4c35-983e-3760c24d65a0.png)

Though this isn't the webpage scrollbar - the width of this scrollbar I have seen on many websites for their whole webpage. Hopefully gives you some insight.

(If I come across another website with a better example, will comment below).

**Comment by summercms** [comment]
Did a quick search in source code search engine and found an example.

Example Website: https://paytm.com/

![image](https://user-images.githubusercontent.com/57409060/134680456-550c9357-4864-4a50-b5a1-72a09a10618b.png)

**Comment by summercms** [comment]
Come across another website today, link here: https://app.santiment.net/

![image](https://user-images.githubusercontent.com/57409060/134770646-0ec95cfb-997b-4ca0-b84f-839e1dc617a1.png)

Pretty much see on a daily basis websites with this issue now (think the problem is epidemic on the web).

**Comment by summercms** [comment]
Came across a different issue with a scrollbar - I thought I'd share!

![image](https://user-images.githubusercontent.com/57409060/135094817-4373f70a-df68-45bd-9287-b3ac92b407f4.png)

The scrollbar on the website I thought didn't exist due to it being almost the same colour as the web page.

The CSS Spec should also include WCAG 2.0 level AA requirements.

**Comment by gregwhitworth** [comment]
Thanks @summercms - this change will need to occur in the CSSWG and so I'm closing this issue.
